api/v1/controllers/samples.js

```diff
@@ -79,6 +82,46 @@ function doFindSample(req, res, next, resultObj, cacheKey, cacheExpiry) {
   });
 } // doFindSample
 
+/**
+ * Avoid non-running Collectors to send in Samples.
+ *
+ * Rejects if it's a Generator request (header.IsGenerator) and its current
+ * Collector has status != RUNNING - response back as 403 Forbidden.
+ *
+ * Resolves when it's a valid Generator or request is not from a Generator.
+ *
+ * @param req
+ * @returns {Promise}
+ */
+function validateNonRunningCollectors(req) {
+  return new Promise((resolve, reject) => {
+    if (!req.headers.IsGenerator) return resolve();
+
+    const param = { key: { value: req.headers.TokenName } };
+    u.findByKey(generators, param)
+    .then((generator) => {
+      if (!generator.currentCollector) {
```

Comment:
I think this needs a "if (generator)" before line 103 so you can handle situation where generator is not found in db